app/gateways/http/authentication/create_token_test.go

```diff
@@ -0,0 +1,134 @@
+			},
+		}, nil)
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader(requestBody))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.CreateToken).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid fields"}`
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+		assert.Equal(t, JsonContentType, response.Header().Get("Content-Type"))
+	})
+
+	t.Run("should return 400 and error when body is empty", func(t *testing.T) {
+		body := LoginRequest{
+			CPF:    "",
+			Secret: "",
+		}
+		requestBody, _ := json.Marshal(body)
+		handler := NewHandler(&authentication.UseCaseMock{
+			CheckLoginFunc:     nil,
+			CreateTokenFunc: func(login authentication.LoginInputs) (string, error) {
+				return "", errors.New("invalid fields")
+			},
+		}, nil)
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader(requestBody))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.CreateToken).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid fields"}`
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+		assert.Equal(t, JsonContentType, response.Header().Get("Content-Type"))
+	})
+
+	t.Run("should return 400 and error when is missing fields", func(t *testing.T) {
+		handler := NewHandler(&authentication.UseCaseMock{
+			CheckLoginFunc:     nil,
+			CreateTokenFunc: func(login authentication.LoginInputs) (string, error) {
+				return "", errors.New("invalid request payload")
+			},
+		}, nil)
+		request, _ := http.NewRequest(http.MethodPost, "/login", bytes.NewReader([]byte{}))
+		response := httptest.NewRecorder()
+
+		http.HandlerFunc(handler.CreateToken).ServeHTTP(response, request)
+
+		got := response.Body.String()
+		expected := `{"error":"invalid request payload"}`
+
+		assert.Equal(t, http.StatusBadRequest, response.Code)
+		assert.Equal(t, expected, strings.TrimSpace(got))
+		assert.Equal(t, JsonContentType, response.Header().Get("Content-Type"))
+	})
+}
+
+func createFakeHandler() Handler {
+	return NewHandler(&authentication.UseCaseMock{
+		CheckLoginFunc: func(inputs authentication.LoginInputs) (bool, error) {
+			return true, nil
+		},
+		CreateTokenFunc: func(login authentication.LoginInputs) (string, error) {
+			return "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOiIwMDAxLTAxLTAxVDAwOjE1OjAwWiIsImlhdCI6IjAwMDEtMDEtMDFUMDA6MDA6MDBaIiwiaWQiOiIxODQxZmM1MC1lMjEwLTQyZmYtYmY2YS0wZTIyNjNjYmJiNGUiLCJpc3MiOiJKV1QiLCJuYW1lIjoiUGF0cmljaWEiLCJzdWIiOiI4ZWI3Zjc5Ny1kZWNiLTQ5NTctOGYxOC01NjU4OTA2ODU4MDgifQ.PbwYkUqGlxdb_pjzAEf0dHacc3eaY5C-sCmpEqXzrwo", nil
+		},
+	}, &account.UseCaseMock{GetByCpfFunc: func(cpf string) (entities.Account, error) {
+		return entities.Account{
+			CPF:       "12345678910",
+			Secret:    "foobar",
+			CreatedAt: time.Time{},
+		}, nil
+	}})
+}
```

Comment:
you use this same fake handler creation logic all along with your code, what do you think to parametrize this function to receive errors and possible returns in a way to make possible flow control for cases where you want errors and the ones you don't want?